[Bug-10394] [Update Possword] Change password shows error

syyangs799 opened the pull request:
…teraction

<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request
This pull request add a status code for password verification length to improve user interaction 
Fixed  [ #10394 ](https://github.com/apache/dolphinscheduler/issues/10394#issue-1266902762) 
<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

Commit:
Add a status code for password verification length to improve user interaction

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:
```diff
@@ -414,7 +414,7 @@ public Map<String, Object> updateUser(User loginUser, int userId,
         if (StringUtils.isNotEmpty(userPassword)) {
             if (!CheckUtils.checkPassword(userPassword)) {
-                putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, userPassword);
+                putMsg(result, Status.USER_PASSWORD_NOT_VALID_OR_TOO_LONG_ERROR, userPassword);
```
please check separately and return different tips.

caishunfeng commented:
Hi @syyangs799 please link the issue to this pr description, such as this pr close #issueId.

Commit:
Modify the length and null value, verify them separately and return

Commit:
Modify the length and null value, verify them separately and return

syyangs799 commented:
> Hi @syyangs799 please link the issue to this pr description, such as this pr close #issueId.

@caishunfeng ok I have modified it to different tips and added corresponding tests

zhuangchong commented:
@syyangs799 Please resolve the conflicting code.

Commit:
Merge remote-tracking branch 'upstream/dev' into dev_issues#10394

# Conflicts:
#	dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java

syyangs799 commented:
> @syyangs799 Please resolve the conflicting code.

Okay, I'm done

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:
```diff
@@ -413,10 +414,15 @@ public Map<String, Object> updateUser(User loginUser, int userId,
         if (StringUtils.isNotEmpty(userPassword)) {
-            if (!CheckUtils.checkPassword(userPassword)) {
+
+            if (!StringUtils.isEmpty(userPassword)) {
                 putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, userPassword);
                 return result;
             }
+            if (CheckUtils.checkPasswordLength(userPassword)) {
+                putMsg(result, Status.USER_PASSWORD_LENGTH_ERROR);
+                return result;
+            }
```
```suggestion
            if (StringUtils.isEmpty(userPassword)) {
                putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, userPassword);
                return result;
            }
            if (CheckUtils.checkPasswordLength(userPassword)) {
                putMsg(result, Status.USER_PASSWORD_LENGTH_ERROR);
                return result;
            }
```

caishunfeng commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:
```diff
@@ -413,10 +414,15 @@ public Map<String, Object> updateUser(User loginUser, int userId,
         }
 
         if (StringUtils.isNotEmpty(userPassword)) {
```
remove the outer not empty validate

syyangs799 replied:
> remove the outer not empty validate

I found that my conditional judgment was wrong. I was very confused about the other judgment. I didn't rule out null values, but I didn't modify it to maintain consistency

caishunfeng replied:
> > remove the outer not empty validate
> 
> I found that my conditional judgment was wrong. I was very confused about the other judgment. I didn't rule out null values, but I didn't modify it to maintain consistency

Sorry @syyangs799 , I found that my original comment is wrong, the outer non-null judgment needs to be retained, and will only be replaced when the new password is not null.
In this way, it's enough to keep the `checkPasswordLength` validate when password not empty, what do you think?

syyangs799 replied:
> > > remove the outer not empty validate
> > 
> > 
> > I found that my conditional judgment was wrong. I was very confused about the other judgment. I didn't rule out null values, but I didn't modify it to maintain consistency
> 
> Sorry @syyangs799 , I found that my original comment is wrong, the outer non-null judgment needs to be retained, and will only be replaced when the new password is not null. In this way, it's enough to keep the `checkPasswordLength` validate when password not empty, what do you think?

ok，I understand that this is to modify the user's other non password information

Commit:
remove the outer not empty validate and Modify judgment conditions

Commit:
add the outer not empty validate and remove inter empty

caishunfeng reviewed (approved):
LGTM :+1:

syyangs799 commented:
> LGTM 👍

@caishunfeng Thank you for my first PR  o(￣▽￣)ｄ

syyangs799 commented:
@caishunfeng Sorry, I clicked the wrong button

zhuangchong reviewed (approved):
LGTM.

caishunfeng commented:
> > LGTM +1
> 
> @caishunfeng Thank you for my first PR o(￣▽￣)ｄ

Thanks for your contribution, looking forward to your next pr :smile:
